e2fsprogs-1.41.12-tests-add-new-test-f_eofblocks.patch

From 8a15d3eadd8ef61d468682f4db447358cbf4dac0 Mon Sep 17 00:00:00 2001
From: Theodore Ts'o <tytso@mit.edu>
Date: Tue, 10 Apr 2012 23:50:37 -0400
Subject: [PATCH 2/2] tests: add new test f_eofblocks

Since we have changed the eofblocks handling in e2fsck (by removing
the need for the EOFBLOCKS_FL flag), we should have a test to check
out how we handle uninitialized and initialized blocks which are exist
beyond i_size, with files that have (and don't have) the EOFBLOCKS_FL.

Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
---
 tests/f_eofblocks/expect.1 |   13 +++++++++++++
 tests/f_eofblocks/expect.2 |    7 +++++++
 tests/f_eofblocks/image.gz |  Bin 0 -> 1364 bytes
 tests/f_eofblocks/name     |    1 +
 4 files changed, 21 insertions(+), 0 deletions(-)
 create mode 100644 tests/f_eofblocks/expect.1
 create mode 100644 tests/f_eofblocks/expect.2
 create mode 100644 tests/f_eofblocks/image.gz
 create mode 100644 tests/f_eofblocks/name

diff --git a/tests/f_eofblocks/expect.1 b/tests/f_eofblocks/expect.1
new file mode 100644
index 0000000..2e9133e
--- /dev/null
+++ b/tests/f_eofblocks/expect.1
@@ -0,0 +1,13 @@
+Pass 1: Checking inodes, blocks, and sizes
+Inode 30, i_size is 2048, should be 4096.  Fix? yes
+
+Inode 31, i_size is 2048, should be 4096.  Fix? yes
+
+Pass 2: Checking directory structure
+Pass 3: Checking directory connectivity
+Pass 4: Checking reference counts
+Pass 5: Checking group summary information
+
+test_filesys: ***** FILE SYSTEM WAS MODIFIED *****
+test_filesys: 31/56 files (29.0% non-contiguous), 83/400 blocks
+Exit status is 1
diff --git a/tests/f_eofblocks/expect.2 b/tests/f_eofblocks/expect.2
new file mode 100644
index 0000000..2a2bca5
--- /dev/null
+++ b/tests/f_eofblocks/expect.2
@@ -0,0 +1,7 @@
+Pass 1: Checking inodes, blocks, and sizes
+Pass 2: Checking directory structure
+Pass 3: Checking directory connectivity
+Pass 4: Checking reference counts
+Pass 5: Checking group summary information
+test_filesys: 31/56 files (29.0% non-contiguous), 83/400 blocks
+Exit status is 0
diff --git a/tests/f_eofblocks/name b/tests/f_eofblocks/name
new file mode 100644
index 0000000..d93adc1
--- /dev/null
+++ b/tests/f_eofblocks/name
@@ -0,0 +1 @@
+blocks after i_size
-- 1.7.7.6


